fil_fungible_token/src/token/mod.rs

```diff
@@ -176,28 +197,32 @@ where
     /// Returns the balance associated with a particular address
     ///
     /// Accounts that have never received transfers implicitly have a zero-balance
-    pub fn balance_of(&self, holder: ActorID) -> Result<TokenAmount> {
+    pub fn balance_of(&self, holder: &Address) -> Result<TokenAmount> {
         // Load the HAMT holding balances
+        let holder = self.resolve_to_id_addr(holder)?;
         Ok(self.state.get_balance(&self.bs, holder)?)
     }
 
     /// Gets the allowance between owner and spender
     ///
     /// The allowance is the amount that the spender can transfer or burn out of the owner's account
     /// via the `transfer_from` and `burn_from` methods.
-    pub fn allowance(&self, owner: ActorID, spender: ActorID) -> Result<TokenAmount> {
+    pub fn allowance(&self, owner: &Address, spender: &Address) -> Result<TokenAmount> {
+        let owner = self.resolve_to_id_addr(owner)?;
+        let spender = self.resolve_to_id_addr(spender)?;
```

Comment:
`get_id_addr` both of these for similar reason. increase_allowance will have initialised if it was called, otherwise allowance must be zero.